k8s/k8s-crd/src/main.rs

```diff
@@ -0,0 +1,82 @@
+#![deny(rust_2018_idioms)]
+#![warn(clippy::all, clippy::pedantic)]
+#![allow(
+    clippy::let_and_return,
+    clippy::let_unit_value,
+    clippy::missing_errors_doc,
+    clippy::too_many_arguments
+)]
+
+use k8s_openapi::api::core::v1::{Event, EventSource, ObjectReference};
+use k8s_openapi::chrono::Utc;
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::MicroTime;
+use kube::api::PostParams;
+use kube::core::ObjectMeta;
+use uuid::Uuid;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let client = kube::Client::try_default().await?;
+    let events_client: kube::Api<Event> = kube::Api::default_namespaced(client);
+
+    let uuid = Uuid::new_v4();
+    let namespace = "default";
+    let resource_version = "1.0.0";
+    let pod_name = "drw-pod";
+
+    let my_event = Event {
+        type_: Some(String::from("Normal")),
+        action: Some(String::from("Ran sample")),
+        count: Some(1),
+        event_time: Some(MicroTime(Utc::now())),
+        first_timestamp: None,
+        last_timestamp: None,
+        reason: Some(String::from("drw process ran")),
+        message: Some(String::from("drw event sample container started")),
+        related: None,
+        reporting_component: Some(String::from("drw-component")),
+        reporting_instance: Some(String::from("k8s-crd")),
+        series: None,
+        source: Some(EventSource {
+            component: Some(String::from("drw-component")),
+            host: Some(String::from("drw-host1")),
+        }),
+        involved_object: ObjectReference {
+            api_version: Some(String::from("v1")),
+            field_path: None,
```

Comment:
From docs:

> If referring to a piece of an object instead of an entire object, this string should contain a valid JSON/Go field access statement, such as desiredState.manifest.containers\[2\]. For example, if the object reference is to a container within a pod, this would take on a value like: "spec.containers{name}" (where "name" refers to the name of the container that triggered the event) or if no container name is specified "spec.containers\[2\]" (container with index 2 in this pod). This syntax is chosen only to have some well-defined way of referencing a part of an object.